src/core/qgsmaplayer.h

```diff
@@ -1499,9 +1499,6 @@ class CORE_EXPORT QgsMapLayer : public QObject
     //! List of layers that may modify this layer on modification
     QSet<QgsMapLayerDependency> mDependencies;
 
-    //! Checks whether a new set of dependencies will introduce a cycle
-    bool hasDependencyCycle( const QSet<QgsMapLayerDependency> &layers ) const;
```

Comment:
Instead of removing this and breaking API, I'd suggest leaving it, marking it as deprecated, and always returning false

Reply:
OK 8983695